sanity-io/structured-content-2022
# Mobile header logo

# Mobile header logo

## Intent

Show the header logo at the top of the page in "mobile" screen sizes, for pages other than the front page.

## Description

As indicated in the Figma sketch for the About page.

I had to rework the menu styling (and markup) quite a bit, because the mobile menu used to consist of the entire `.menuContents`, where some items were hidden. But now the contents are really positioned completely separately: the logo/`.homeLink` is at the top of the page in normal flow, whereas the menu toggle button is fixed-positioned near the bottom of the viewport, and the menu items are also `position: fixed`.

As part of this refactoring, the two separate buttons for "Menu" vs "Close" are now also merged into one single on/off toggle button (whose appearance mirrors the old buttons, depending on the open/close state).

The `Nav` is also now told whether or not it's on the front page, so it can know whether or not the `.homeLink` should be showing on mobile.

Summary of the design "spec" for the header/Nav:
- On tablet/desktop:
    - On the front page: Nav is completely hidden initially, appears fixed at the top of the viewport when page is scrolled down past a certain point
    - On other pages: Nav is sticky at the top of the viewport at all times
- On mobile: Menu button is always fixed near the bottom of the viewport
    - On the front page: Home link/logo is always hidden
    - On other pages: Home link/logo is showing at the top of the page, not fixed/sticky (can be scrolled away)

## Testing this PR
1. Open the [About page](https://structured-content-2022-web-git-mobile-header-logo.sanity.build/about)
2. Shrink the window to <700px width, or use "Responsive Design Mode" or equivalent feature in your dev tools with some phone-ish form factor (in portrait orientation)
3. Verify that the "Structured Content 2022" logo appears at the top of the page, and scrolls away when the page is scrolled down
